File: src/appDatabase.js
import { MongoClient } from "mongodb"
import 'dotenv/config'


const mongoClient = new MongoClient(process.env.MONGO_HOST)
const database = mongoClient.db(process.env.MONGO_DB)
const collection = database.collection(process.env.MONGO_COLLECTION)


export async function initDatabase() {
    await mongoClient.connect()
    console.log('Connection with MongoDB established.')
}


export async function insertMessageIntoDB(message) {
    await collection.insertOne(JSON.parse(message))
}


export function closeDatabase() {
    mongoClient.close()
}

File: src/main/java/AppDatabase.java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import org.bson.Document;

public class AppDatabase {
    private final MongoCollection<Document> mongoCollection;

    public AppDatabase(String hostname, String database, String collection) {
        MongoClient mongoClient = MongoClients.create(hostname);
        this.mongoCollection = mongoClient
                .getDatabase(database)
                .getCollection(collection);
    }

    public void insertMessage(String stringifiedJsonMessage) {
        Document parsedBsonMessage = Document.parse(stringifiedJsonMessage);
        mongoCollection.insertOne(parsedBsonMessage);
    }
}
